File: backend/src/main/java/io/github/artemy/osipov/shop/service/candy/CandyRepository.java
package io.github.artemy.osipov.shop.service.candy;

import io.github.artemy.osipov.shop.exception.EntityNotFoundException;
import org.springframework.data.mongodb.repository.MongoRepository;

import java.util.UUID;

public interface CandyRepository extends MongoRepository<Candy, UUID> {

    default Candy getById(UUID id) {
        return findById(id)
                .orElseThrow(() -> new EntityNotFoundException(Candy.class, id));
    }
}

File: backend/src/main/kotlin/io/github/artemy/osipov/shop/service/candy/CandyRepository.kt
package io.github.artemy.osipov.shop.service.candy

import io.github.artemy.osipov.shop.exception.EntityNotFoundException
import io.github.artemy.osipov.shop.repository.ExtraMethods
import org.springframework.data.repository.kotlin.CoroutineCrudRepository
import java.util.*

interface CandyRepository : CoroutineCrudRepository<Candy, UUID>, ExtraMethods<Candy> {

    companion object {
        suspend fun CandyRepository.getById(id: UUID): Candy {
            return findById(id)
                ?: throw EntityNotFoundException(Candy::class, id)
        }
    }
}
